### PR TITLE
🏗 Flag an error if `--files` matches zero test files

### PR DESCRIPTION
**Pull request by rsimha**
This PR adds a useful message if `amp [unit|integration|e2e] --files` is called with non-existent test files.

**Before:**

image

**After:**

image

### CODE REVIEW

File: build-system/common/utils.js [hunk 1]
```diff
@@ -96,21 +96,24 @@ function logFiles(files) {
...
   const toPosix = (str) => str.replace(/\\\\?/g, '/');
-  return argv.files
-    ? globby.sync(
-        (Array.isArray(argv.files) ? argv.files : argv.files.split(','))
-          .map((s) => s.trim())
-          .map(toPosix)
-      )
-    : [];
+  const globs = Array.isArray(argv.files) ? argv.files : argv.files.split(',');
+  const files = globby.sync(globs.map((glob) => glob.trim()).map(toPosix));
```
Comment (rcebulko) [inline build-system/common/utils.js:111]:
```suggestion
  const files = globby.sync(globs.map((glob) => toPosix(glob.trim())));
```

File: build-system/common/utils.js [hunk 2]
```diff
@@ -96,21 +96,24 @@ function logFiles(files) {
...
+  if (files.length == 0) {
+    log(red('ERROR:'), 'Argument(s)', cyan(argv.files), 'matched zero files.');
+    throw new Error('No tests to run.');
```
Comment (rcebulko) [inline build-system/common/utils.js:114]:
```suggestion
    throw new Error('No test files found matching specified file pattern.');
```

**Review by rcebulko — APPROVED** [review]
Approving to unblock

File: build-system/common/utils.js (hunk 1 above)
Comment (rsimha) [reply build-system/common/utils.js]: Done. 👍

File: build-system/common/utils.js (hunk 2 above)
Comment (rsimha) [reply build-system/common/utils.js:114]: This info is already contained in the error message on line 113.

**Review by kristoferbaxter — APPROVED** [review]

File: build-system/common/utils.js (hunk 2 above)
Comment (rcebulko) [reply build-system/common/utils.js:114]: Many places in `build-system` which explicitly `log` an error message then use the same string (or a replicated version without colors) as the message for the error thrown. I think that makes more sense here. Saying "no tests to run" has a different implication.

Examples:
https://github.com/ampproject/amphtml/blob/main/build-system/tasks/runtime-test/runtime-test-base.js#L402-L404
https://github.com/ampproject/amphtml/blob/main/build-system/tasks/check-sourcemaps.js#L58-L157

Comment (rsimha) [reply build-system/common/utils.js]: Fair enough. I've used the same message without the (potentially long) list of args.

File: build-system/common/utils.js [hunk 3]
```diff
@@ -96,21 +96,24 @@ function logFiles(files) {
...
   const toPosix = (str) => str.replace(/\\\\?/g, '/');
-  return argv.files
-    ? globby.sync(
-        (Array.isArray(argv.files) ? argv.files : argv.files.split(','))
-          .map((s) => s.trim())
-          .map(toPosix)
-      )
-    : [];
+  const globs = Array.isArray(argv.files) ? argv.files : argv.files.split(',');
+  const files = globby.sync(globs.map((glob) => toPosix(glob.trim())));
```
Comment (jridgewell) [inline build-system/common/utils.js:111]: Nit: Maybe issue multiple requests to globby, and check if any pattern doesn't match? Eg, `unit/foo/test-*.js, non-existent/test-*.js` should probably warn, too.

**Review by jridgewell — APPROVED** [review]

File: build-system/common/utils.js (hunk 3 above)
Comment (rsimha) [reply build-system/common/utils.js]: Nice catch. Updated.

**Pull request merged**